### PR TITLE
natspecify + docs

### PR DESCRIPTION
**Pull request by Flip-Liquid**
Adds natspec style documentation to the core smart contracts

### CODE REVIEW

File: src/interfaces/IOptionSettlementEngine.sol [hunk 1]
```diff
@@ -119,52 +195,115 @@ interface IOptionSettlementEngine {
...
+     * each address in tokens.
+     * @param tokens The tokens for which fees will be swept to the feeTo address.
+     */
     function sweepFees(address[] memory tokens) external;
 
-    // @notice Create a new options chain from optionInfo if it doesn't already exist
+    /**
+     * @notice Create a new options chain from optionInfo if it doesn't already exist
+     * @param optionInfo The optionInfo from which a new chain will be created, with
+     * settlementSeed set to 0.
```
Comment (0xAlcibiades) [inline src/interfaces/IOptionSettlementEngine.sol:275]: Is it set to 0, or is it set to the hash of the option?

**Review by 0xAlcibiades — CHANGES_REQUESTED** [review]
Only one small thing needs to change here.

File: src/interfaces/IOptionSettlementEngine.sol [hunk 2]
```diff
@@ -4,54 +4,94 @@ pragma solidity ^0.8.0;
...
+    /**
+     * @notice The optionId specified expired at expiry.
+     * @param optionId The id of the expired option.
+     * @param expiry The time of expiry of the supplied option Id.
+     */
     error ExpiredOption(uint256 optionId, uint40 expiry);
 
-    // This option cannot yet be exercised.
+    /// @notice This option cannot yet be exercised.
     error ExerciseTooEarly();
```
Comment (thal0x) [inline src/interfaces/IOptionSettlementEngine.sol:62]: I know this is just documentation updates but there should probably be an associated optionId param for this event.

File: src/interfaces/IOptionSettlementEngine.sol [hunk 3]
```diff
@@ -119,52 +195,115 @@ interface IOptionSettlementEngine {
...
+     * @notice Returns the token type (e.g. Option/Claim) for a given token Id
+     * @param tokenId The id of the option or claim.
+     * @return The enum (uint8) Type of the tokenId
+     */
     function tokenType(uint256 tokenId) external view returns (Type);
 
-    // @return The optionInfo Option struct for tokenId
+    /**
+     * @notice Returns Option struct details about a given tokenID if that token is
+     * a vToken.
```
Comment (thal0x) [inline src/interfaces/IOptionSettlementEngine.sol:237]: Is vToken a thing? This is the first time i'm seeing it. Maybe a definition on what exactly that is would be helpful for future readers.

File: src/interfaces/IOptionSettlementEngine.sol (hunk 2 above)
Comment (0xAlcibiades) [reply src/interfaces/IOptionSettlementEngine.sol:62]: Do you want expiry on that as well?

File: src/interfaces/IOptionSettlementEngine.sol (hunk 3 above)
Comment (0xAlcibiades) [reply src/interfaces/IOptionSettlementEngine.sol:237]: Same here, new term to me. @Flip-Liquid what do you mean here by vToken?

**Comment by 0xAlcibiades** [comment]
Re proposed event changes, let's get an issue and a separate PR going @thal0x

File: src/interfaces/IOptionSettlementEngine.sol (hunk 3 above)
Comment (0xAlcibiades) [reply src/interfaces/IOptionSettlementEngine.sol:237]: Heh, new in that I forgot about it, apparently. I think I meant "valid valorem token" i.e. not None type.

Comment (Flip-Liquid) [reply src/interfaces/IOptionSettlementEngine.sol:237]: fixed

File: src/interfaces/IOptionSettlementEngine.sol (hunk 1 above)
Comment (Flip-Liquid) [reply src/interfaces/IOptionSettlementEngine.sol:275]: The resultant chainKey is the output of the kecak256 hash of the option with the seed set to zero. But internally, it's set to the hash of the option. I can clarify this in the comment.

Comment (Flip-Liquid) [reply src/interfaces/IOptionSettlementEngine.sol]: @0xAlcibiades updated, lmk if that new comment clarifies it sufficiently.

File: src/interfaces/IOptionSettlementEngine.sol [hunk 4]
```diff
@@ -234,7 +234,7 @@ interface IOptionSettlementEngine {
 
     /**
      * @notice Returns Option struct details about a given tokenID if that token is
-     * a vToken.
+     * a option.
```
Comment (0xAlcibiades) [inline src/interfaces/IOptionSettlementEngine.sol:237]: an :)

**Review by 0xAlcibiades — APPROVED** [review]

**Pull request merged**